skl/skl.go

```diff
@@ -358,6 +359,110 @@ func (s *Skiplist) Put(key []byte, v y.ValueStruct) {
 	}
 }
 
+// Hint is used to speed up sequential write.
+type Hint struct {
+	height int32
+	prev   [maxHeight + 1]*node
+	next   [maxHeight + 1]*node
+}
+
+// PutWithHint inserts the key-value pair with Hint for better sequential write performance.
+func (s *Skiplist) PutWithHint(key []byte, v y.ValueStruct, hint *Hint) {
```

Comment:
It's easy to make mistake, we use the PutWithHint method to implement Put method with nil hint pointer.

Reply:
ok